### PR TITLE
Vampire species for halloween

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Added vampires. They will be available as a roundstart race during the Halloween holiday event.
/:cl:

Vampires are another halloween species. 

They slowly lose blood constantly (unless in a coffin). They dust if they run out of blood.

They heal in coffins (but do not restore blood)

They can drain blood from other mobs they have in a passive grab.

They can turn into a bat.

They catch on fire if they enter the chapel.

Why: The more variety for halloween races the merrier. Having the background pressure of keeping your blood level up and finding "donors" without getting spaced by the AI might be fun as a limited time gimmick. This is also, bizarrely, one of the more commonly requested features as well.

### CODE REVIEW

**Comment by Dicks1** [comment]
So I told Kor to add vampires for Halloween.... he actually did it the absolute madman
hahahahahahahahaha!

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+/datum/action/item_action/organ_action/vampire/Trigger()
+	. = ..()
+	if(iscarbon(owner))
+		var/mob/living/carbon/H = owner
+		if(H.pulling && iscarbon(H.pulling))
+			var/mob/living/carbon/victim = H.pulling
+			if(victim.stat == DEAD)
+				to_chat(H, "<span class='notice'>You need a living victim!</span>")
+				return
+			if(victim.blood_volume < 50)
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: this should be covered by the above check

Comment (KorPhaeron) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: For whatever reason the check for instant blood volume death is only for humans

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	C.update_body(0)
+	var/obj/effect/proc_holder/spell/targeted/shapeshift/bat/B = new
+	C.AddSpell(B)
+
+/datum/species/vampire/spec_life(mob/living/carbon/human/C)
+	. = ..()
+	if(istype(C.loc, /obj/structure/closet/coffin))
+		C.heal_overall_damage(4,4)
+		C.adjustToxLoss(-4)
+		C.adjustOxyLoss(-4)
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: clone loss, too?

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		return
+	C.blood_volume -= 1.5
+	if(C.blood_volume <= 0)
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: suffering from actual blood loss effects means this check won't trigger

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			victim.blood_volume = Clamp(victim.blood_volume - 50, 0, BLOOD_VOLUME_MAXIMUM)
+			H.blood_volume = Clamp(H.blood_volume + 50, 0, BLOOD_VOLUME_MAXIMUM)
+
+/obj/effect/proc_holder/spell/targeted/shapeshift/bat
+	name = "Bat Form"
+	desc = "Take on the shape a space bat."
+	invocation = "Squeak!"
+
+	shapeshift_type = /mob/living/simple_animal/hostile/retaliate/bat
+	list/current_shapes = list(/mob/living/simple_animal/hostile/retaliate/bat)
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: remove these `list/`s

**Comment by JJRcop** [comment]
Make sure vampires can't be chaplains.

**Comment by Bawhoppen** [comment]
Where's the garlic, holy water, and wooden stakes weaknesses?

**Comment by DoctoreMad** [comment]
>Garlic

Can we let our vampires not degrade into Friday the 13th tier?

**Comment by Armhulen** [comment]
testmergeable?

**Comment by Armhulen** [comment]
holy water should burn them, make humans rarely start as vampires (not an antag, just have to get blood to survive good luck!!

**Comment by Grazyn** [comment]
I'm not convinved about them being roundstart, even for a limited 5-day event. How does this work rule-wise exactly? You literally have to harm people to survive, people who will obviously fight back or even kill you preventively, does this mean vampires are always valid? If you're a vampire and not antag, can you get banned for killing/incapacitating someone to feed yourself? If not, it means that being a vampire counts as being antag, and what's stopping every single player on the server from signing up as one?

**Comment by JJRcop** [comment]
@Grayzn Doesn't make them an antag, and it doesn't make them valid. If you kill someone as a vamp, then that's excessive, since you can incapacitate them for a short time, or just find someone else if they put up too much of a fuss. If you are desperately running out of blood and you can't catch someone alive long enough to suck their blood, you just die, like plasmamen running out of plasma.

I don't think vamps get a free pass to kill by any means, you just need to drink blood to survive, and you can do that while still staying within the server's rules.

**Comment by Grazyn** [comment]
Well I guess we'll see how it plays out, but keep in mind that currently any fight can easily escalate from throwing punches to someone ending up spaced. As non-antag vampire, you either find a willing volunteer or you must be prepared to die, because starting a fight with your victim means they are cleared to kill you while you can't do the same to them (after all, how can your victim know you're not gonna drain them dry once you have them stunlocked). I can't really picture an admin banning someone for killing a vampire who attacked them.

**Comment by Grazyn** [comment]
Or let me put it this way: would you use lethal damage on someone who attacked you and cut off your leg? Do you think an admin would ban you for using lethal damage in that situation? Because the inconvenience you get from being low on blood is the same if not worse than having one leg cut off. You constantly pass out, you take oxy damage which will add to your preexisting damage, meaning you can easily end up in crit, you need to go to medbay to heal it and it will take some time, either with transfusion or iron.

Saying "You can kill a vamp if an IC fight has been properly escalated" is disingenuous, because the attempt itself will be enough to make the vamp valid.

**Comment by Bawhoppen** [comment]
People could also not be dumb, that would probably solve it too

**Comment by Grazyn** [comment]
The event will run for 5 days, this means hundreds of rounds, we can either address this issue now or deal with it then. "Hahahaha just not be dumb like nigga just let the vamp drain you haha" is not really addressing it.

**Comment by JJRcop** [comment]
When I shapeshifted into a bat I couldn't change back.

**Comment by pantherpusher** [comment]
What if people got greentext for "donating" blood to the vampire crew?

**Comment by KorPhaeron** [comment]
@jjrcop

That would make sense since I gave the spell to the vampire mob

Whoops

**Comment by Hatfish** [comment]
Keep up the masquerade or get dunked, it's not hard to understand neonates.

**Comment by davethwave** [comment]
Also as a vampire is not a antag try to add some roundstart text saying so to prevent a load of self-antags starting on tg due to the confusion between this Halloween vampire and the antag one.

**Comment by Armhulen** [comment]
rarely off of halloween humans start as vampires when??

**Comment by JJRcop** [comment]
KorPhaeron#30

I gave vampires the ability to drink blood (the reagent) to restore their blood_volume

**Comment by Ratvar** [comment]
Bloodtomato masterrace

**Comment by DoctoreMad** [comment]
@Armhulen This feature is purely for fun, not to annoy people. I'd hate to be a vampire xenobiologist.

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+			return
+		if(H.pulling && iscarbon(H.pulling))
+			var/mob/living/carbon/victim = H.pulling
+			if(victim.stat == DEAD)
+				to_chat(H, "<span class='notice'>You need a living victim!</span>")
+				return
+			if(victim.dna && ((NOBLOOD in victim.dna.species.species_traits) || victim.dna.species.exotic_bloodtype))
+				to_chat(H, "<span class='notice'>Your victim doesn't have blood!</span>")
+				return
+			if(victim.blood_volume < 50)
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: bit of an arbitrary limit.
could just say "you didn't get a full meal!" and pull all thats left.

**Comment by Repukan** [comment]
Do they have the ability to stop time?

**Comment by Armhulen** [comment]
@DoctoreMad get good, healing from coffins is pretty good

File: code/modules/mob/living/carbon/human/species_types/vampire.dm (hunk 2 above)
Comment (KorPhaeron) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: This got added

File: code/modules/mob/living/carbon/human/species_types/vampire.dm (hunk 5 above)
Comment (KorPhaeron) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: @JJRcop could you do this please? Simple but I will not have the chance today

Comment (KorPhaeron) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: Also change the bat spell to try to grant to mind if you have the time

**Comment by KorPhaeron** [comment]
@armhulen

Healing from coffins (and turning into a bat) are also why I think this is too silly and off theme outside of halloween.

**Comment by Durkel** [comment]
If the chaplains whip nullrod variant doesn't do bonus damage against vampires I will never have respect for you again.

**Comment by DoctoreMad** [comment]
Demoniac race for Halloween when
Just a human that is possessed by a demon.

**Comment by Repukan** [comment]
Make them take extra damage from holoparasites.

**Comment by Bawhoppen** [comment]
>assistant breaks into medbay blood storage
>gets caught and tries to run
>"HE MUST BE A VAMP! GET HIM!"

sounds fun actually

File: code/modules/mob/living/carbon/human/species_types/vampire.dm (hunk 5 above)
Comment (JJRcop) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: Sure

**Comment by davethwave** [comment]
remember to make them stronger than the average person too and perhaps have a minor upgrade relating to robustness or something like that, just make sure that at least give a upside or 2 to the vampire since for instance, by turning into a bat what help is it besides having a smaller hit box and perhaps being a bit faster.

File: code/modules/mob/living/carbon/human/species_types/vampire.dm (hunk 5 above)
Comment (KorPhaeron) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: And maybe change V blood to U for Universal blood so it feels less snowflake

Comment (KorPhaeron) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm]: Nevermind I found time to do this

**Comment by davethwave** [comment]
Okay I retract my statement about the bat but still, a lot of downside not a lot of up.

**Comment by KorPhaeron** [comment]
@davethwave free regen in coffins, transform spell that lets you ventcrawl, and night vision

Might already be too much for roundstart rather than too little honestly

**Comment by davethwave** [comment]
wait, bats can ventcrawl?

**Comment by KorPhaeron** [comment]
Yes

They can move in space as well

**Comment by KorPhaeron** [comment]
Honestly I should probably remove night vision from them since they have no other interaction with light anyway

**Comment by davethwave** [comment]
then I retract all of my other statements about the vampire not being powerful, the power to vent crawl is already op if used properly.

**Comment by JJRcop** [comment]
Dunno how you will feel about this but KorPhaeron#32

**Comment by GunHog** [comment]
What if people that are succed dry by a vampire, then bitten again become vampires themselves?

**Comment by KorPhaeron** [comment]
@grazyn

I plan to admin it like I would stealing a space suit in a crisis.

If you make a good faith effort to find monkeys/willing donors/sabutamol and you are still about to starve to death it will be okay to eat the clown

If you, roundstart, start eating the clown and kill him when he retaliates that wont be okay

I Am Legend scenarios caused by a traitor destroying medbay and the human heads of staff holed up in the chapel while the vampire crew riots for blood sounds fun

File: code/modules/reagents/reagent_containers/blood_pack.dm [hunk 6]
```diff
@@ -65,6 +65,9 @@
 /obj/item/reagent_containers/blood/lizard
 	blood_type = "L"
 
+/obj/item/reagent_containers/blood/universal
```
Comment (MoreRobustThanYou) [inline code/modules/reagents/reagent_containers/blood_pack.dm]: > universal

WHY. JUST WHY?

**Comment by davethwave** [comment]
And of course there is the mourge as well as cloning, even just asking genetics would probably be fine once the idea of vampire does not equal antag sets in throughout the crew.

**Comment by KorPhaeron** [comment]
@tgstation/commit-access Halloween starts in about 24 hours on /tg/station so I am review/merge begging

**Comment by JJRcop** [comment]
Vamps cån't suhh bluhh from deeed people, althhhough, you could rrrremooove the deeed person's bluhh ussssing an ivvvvyy drrrriiiip

**Comment by FantasticFwoosh** [comment]
This seems like a level of detail that is unnecessary on a starting race for Halloween and more of a side-antagonist/experimental project @KorPhaeron 

Couldn't you tone them back a tiny bit with blood attached to nutrition or a blood subgroup of likable tastes (**including drinking at the bar bloody mary's / salbutomol / tomato soup etc**) and no explicit powers/particularly degrading weaknesses unless they are actually a antagonist?

- Holy water/chaplain weapon weakness, nonconvertible to cult for said reason but otherwise slightly accelerated  nutritional regeneration & a permanent eternal youth status? (as soon as that virus effect actually means anything)

There is a vampire mask floating around on other servers thematically to use if we just wanted to input a vampire antagonist status onto someone.

- Sucking blood out of someone is constituted as a harmful and (valid?) act without consent i guess.

**Pull request merged**